src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -93,6 +96,50 @@
   (instance? X509CRL x))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Schemas
+
+(def ValidX500Name
+  (schema/pred valid-x500-name?))
+
+(def Extension
+  "A map containing all the fields required to define an extension."
+  {:oid schema/Str
+   :critical (schema/pred #(not (nil? %)))
```

Comment:
oh. most (all?) of the functions explicitly cast this to a boolean, so that's probably safe for output. I'll make sure we aren't being lax in input somewhere and change this to `schema/Bool`.